metadata_tools.py

```diff
@@ -120,20 +123,32 @@ def read_exif_metadata(self, convert_tags=True):
         if convert_tags is True:
             exif = {
                 PIL.ExifTags.TAGS[k]: v
-                for k, v in img._getexif().items()
+                for k, v in img.getexif().items()
                 if k in PIL.ExifTags.TAGS
             }
         else:
-            exif = img._getexif()
+            exif = img.getexif()
 
         img.close()
         return exif
 
```

Comment:
function process exif ring, just applies the function exif_attach_metadata to the config exif ring data dictionary.